core/src/tangram.cpp

```diff
@@ -241,6 +247,23 @@ void render() {
     FrameInfo::draw(*m_view, *m_tileManager);
 
     while (Error::hadGlError("Tangram::render()")) {}
+
+
+    bool ready = !(m_view->changedOnLastUpdate() ||
+                   m_tileManager->hasTileSetChanged() ||
+                   m_tileManager->hasLoadingTiles() ||
+                   m_labels->needUpdate());
+    if (ready) {
+        for (auto& ease : m_eases) {
+            if (!ease.finished()) { ready = false; }
+        }
+    }
+
+    m_renderDoneCallback(ready);
```

Comment:
Yeah, the original goal here was a "view complete" event/callback. You're right that it should really only be called when "complete" changes from false to true, and should probably be called something more like "view complete" rather than "render done".